tasks/fork/create-checkpoints.ts

```diff
@@ -0,0 +1,109 @@
+import { task, types } from "hardhat/config";
+import { Address } from "viem";
+import {
+    HyperdriveDeployBaseTask,
+    HyperdriveDeployBaseTaskParams,
+} from "../deploy";
+import { sleep } from "./lib";
+
+export type CreateCheckpointsParams = HyperdriveDeployBaseTaskParams & {
+    pollInterval: number;
+};
+
+export function getCheckpointId(
+    checkpointDuration: bigint,
+    blockTimestamp: bigint,
+) {
+    return blockTimestamp - (blockTimestamp % checkpointDuration);
+}
+
+HyperdriveDeployBaseTask(
+    task(
+        "fork:create-checkpoints",
+        "Long-running process that creates checkpoints for all hyperdrive instances when necessary",
+    ),
+)
+    .addOptionalParam(
+        "pollInterval",
+        "Amount of time in minutes between polls for checkpoint state",
+        240,
+        types.int,
+    )
+    .setAction(
+        async (
+            { pollInterval }: Required<CreateCheckpointsParams>,
+            { viem, artifacts, hyperdriveDeploy, config, network },
+        ) => {
+            let poolAddresses = config.networks[
+                network.name
+            ].hyperdriveDeploy!.instances.map(
+                (instance) =>
+                    hyperdriveDeploy.deployments.byName(instance.name).address,
+            );
+            console.log(`found ${poolAddresses!.length} pools`);
+            let checkpointDurations = !poolAddresses
+                ? []
+                : await Promise.all(
+                      poolAddresses.map(
+                          async (a) =>
+                              (
+                                  await (
+                                      await viem.getContractAt(
+                                          "HyperdriveTarget0",
+                                          a,
+                                      )
+                                  ).read.getPoolConfig()
+                              ).checkpointDuration,
```

Comment:
Why are we reading this from `Target0`? We can just use `IHyperdrive` for this. Maybe this is one of the getters that you were talking about the other day.